tests/sdk/context/condenser/test_condenser_from_spec.py

```diff
@@ -0,0 +1,165 @@
+"""Test CondenserBase.from_spec functionality."""
+
+import pytest
+
+from openhands.sdk.context.condenser.base import CondenserBase
+from openhands.sdk.context.condenser.spec import CondenserSpec
+from openhands.sdk.llm import LLM
+
+
+@pytest.fixture
+def basic_llm():
+    """Create a basic LLM for testing."""
+    return LLM(model="test-model")
+
+
+def test_from_spec_no_op_condenser():
+    """Test creating a NoOpCondenser from spec."""
+    spec = CondenserSpec(name="NoOpCondenser", params={})
+
+    condenser = CondenserBase.from_spec(spec)
+
+    assert isinstance(condenser, CondenserBase)
+    # The from_spec method creates using the base class constructor
+    # which should work for any condenser type
+
+
+def test_from_spec_llm_summarizing_condenser(basic_llm):
+    """Test creating an LLMSummarizingCondenser from spec."""
+    spec = CondenserSpec(
+        name="LLMSummarizingCondenser",
+        params={"llm": basic_llm, "max_size": 80, "keep_first": 10},
+    )
+
+    condenser = CondenserBase.from_spec(spec)
+
+    assert isinstance(condenser, CondenserBase)
+
+
+def test_from_spec_with_empty_params():
+    """Test creating a condenser with empty parameters."""
+    spec = CondenserSpec(name="NoOpCondenser", params={})
+
+    condenser = CondenserBase.from_spec(spec)
+
+    assert isinstance(condenser, CondenserBase)
+
+
+def test_from_spec_with_complex_params(basic_llm):
+    """Test creating a condenser with complex parameters."""
+    spec = CondenserSpec(
+        name="LLMSummarizingCondenser",
+        params={"llm": basic_llm, "max_size": 120, "keep_first": 15},
+    )
+
+    condenser = CondenserBase.from_spec(spec)
+
+    assert isinstance(condenser, CondenserBase)
```

Comment:
@OpenHands you should assert for the actual type across all tests, in this case it should be `LLMSumarizingCondenser`.

Also make sure from_spec also works for PipelineCondenser which requires a condensers as input. You should modify PipelineCondenser so it also accepts list[CondenserSpec] in additional to existing list[Condenser], maybe a field validator